jsonschema/validators.py

```diff
@@ -161,15 +161,22 @@ def iter_errors(self, instance, _schema=None):
             if scope:
                 self.resolver.push_scope(scope)
             try:
-                validators = []
-
-                # We make sure $ref is evaluated first if available in schema
-                ref = _schema.get(u"$ref")
-                if ref is not None:
-                    validators = [(u"$ref", ref)]
-
-                # Add all remaining validators
-                validators += [(x, _schema[x]) for x in _schema if x not in ["$ref"]]
+                if isinstance(self, Draft202012Validator):
```

Comment:
This will have to be updated for every new validator such as 2019-09. That seems error prone. I think the validator needs some sort of flag for this or perhaps a version that can be compared using =, <, or >.

Reply:
I agree. For now I made a list of older classes in `legacy_cls` in the new `schema_validators` method, so newer versions should not be affected.